code/cgame/cg_weapons.c

```diff
@@ -1696,6 +1710,14 @@ void CG_FireWeapon(centity_t *cent) {
 		}
 	}
 
+	// random spray color if spray pistol or injector is used
+	// and random color menu option is enabled
+	if (cgs.clientinfo[cent->currentState.clientNum].randomcolor == 1 &&
+			(ent->weapon == WP_KMA97 || ent->weapon == WP_SPRAYPISTOL)) {
+		int rnd = (int)(random() * ((float)NUM_SPRAYCOLORS - 0.1f));
```

Comment:
here you can use `randomindex(NUM_SPRAYCOLORS)`

Reply:
randomindex(NUM_SPRAYCOLORS -1) I think.